### PR TITLE
LUCENE-9302,SOLR-14381: Integer overflow in total count in grouping results (7x/master)

### PR DESCRIPTION
**Pull request by chatman**
This is based on #54 (which is for 8x). This is part two of the grouping fix, meant to be merged to 7x/master. For reference, the previous (intermediate) fix was reviewed here: https://github.com/fullstorydev/lucene-solr/pull/55

### CODE REVIEW

File: lucene/grouping/src/java/org/apache/lucene/search/grouping/BlockGroupingCollector.java [hunk 1]
```diff
@@ -354,8 +354,8 @@ public BlockGroupingCollector(Sort groupSort, int topNGroups, boolean needsScore
 
     return new TopGroups<>(new TopGroups<>(groupSort.getSort(),
                                        withinGroupSort.getSort(),
-                                       totalHitCount, totalGroupedHitCount, groups, maxScore),
-                         totalGroupCount);
+                                      (long) totalHitCount, (long) totalGroupedHitCount, groups, maxScore),
```
Comment (shanhe-fullstory) [inline lucene/grouping/src/java/org/apache/lucene/search/grouping/BlockGroupingCollector.java:357]: These two casts should happen automatically from int to long.

File: lucene/grouping/src/java/org/apache/lucene/search/grouping/GroupingSearch.java [hunk 2]
```diff
@@ -163,7 +163,7 @@ protected TopGroups groupByFieldOrFunction(IndexSearcher searcher, Query query,
     }
 
     if (allGroups) {
-      return new TopGroups(secondPassCollector.getTopGroups(groupDocsOffset), matchingGroups.size());
+      return new TopGroups(secondPassCollector.getTopGroups(groupDocsOffset), (long) matchingGroups.size());
```
Comment (shanhe-fullstory) [inline lucene/grouping/src/java/org/apache/lucene/search/grouping/GroupingSearch.java:166]: int -> long should happen automatically.

File: solr/core/src/java/org/apache/solr/core/SolrCore.java [hunk 3]
```diff
@@ -2558,6 +2558,9 @@ public void execute(SolrRequestHandler handler, SolrQueryRequest req, SolrQueryR
 
     preDecorateResponse(req, rsp);
 
+    if (req.getOriginalParams().get("group") != null) {
+      System.out.println("hello: "+req);
```
Comment (shanhe-fullstory) [inline solr/core/src/java/org/apache/solr/core/SolrCore.java:2562]: Remove?

**Review by shanhe-fullstory — APPROVED** [review]

File: solr/core/src/java/org/apache/solr/core/SolrCore.java (hunk 3 above)
Comment (chatman) [reply solr/core/src/java/org/apache/solr/core/SolrCore.java:2562]: Ah, good catch.

**Comment by chatman** [comment]
Thanks @shanhe-fullstory!

**Pull request merged**